### DISCUSSION TITLE
System Register (S3_3_C2_C4_0) Compilation Error

### DISCUSSION THREAD

**Discussion by OH-YG** [tianocore/edk2#5821] [Q&A]
Dear

I am working on a UEFI FMP Driver for QualComm Arm
It's my first time building Arm Architecture as a target, and an error occurred at the bottom while using the random number generation library.

Build log as below
[ArmRngError.log](https://github.com/user-attachments/files/15965993/ArmRngError.log)

If I look at the error content, you get an error that says "S3_3_C2_C4_0" should be constant.
What is meaning that "S3_3_C2_C4_0"? And How to fix this error?

EDK PlatForm: **EDK2023_02**
Compiler : **MSVC v142 Arm64**
Clang & SDLLVM Version: **14.0.1**
Development Environment: Windows 10

**Comment by ghost** [comment]
I'd guess the Microsoft compiler doesn't understand that `S3_3_C2_C4_0` syntax refers to a system register.
